sys/arm64/arm64/elf_machdep.c

```diff
@@ -296,11 +314,78 @@ elf_cpu_parse_dynamic(caddr_t loadbase __unused, Elf_Dyn *dynamic __unused)
 
 	return (0);
 }
+
+#ifdef __CHERI_PURE_CAPABILITY__
+/*
+ * Handle boot-time kernel relocations, this is called by locore.
+ */
+void
+elf_boot_dyn_reloc(void *data_cap, const void *code_cap, const Elf_Dyn *dyn)
```

Comment:
PowerPC calls this elf_reloc_self. Also I'd put dyn first, and it tends to be called dynp.

Reply:
Fixed